R/benchmark.R

```diff
@@ -201,7 +203,9 @@ benchmark = function(design, store_models = FALSE, store_backends = TRUE, encaps
 
   set(grid, j = "mode", value = NULL)
 
-  result_data = ResultData$new(grid, store_backends = store_backends)
+  data_extra = if (length(callbacks)) map(res, "data_extra")
```

Comment:
Do we set `data_extra` to `NULL` if there are callbacks but none of them writes `extra_data`?

Reply:
No then `data_extra` is created. I thought that `any(map_lgl(res, function(x) is.null(x$data_extra)))` is much more expensive than `length(callbacks)`.  But maybe it is unreasonable to worry about 1/3 milliseconds overhead.